salt/utils/extend.py

```diff
@@ -0,0 +1,305 @@
+# -*- coding: utf-8 -*-
+'''
+SaltStack Extend
+~~~~~~~~~~~~~~~~
+
+A templating tool for extending SaltStack.
+
+Takes a template directory and merges it into a SaltStack source code
+directory. This tool uses Jinja2 for templating.
+
+This tool is accessed using `salt-extend`
+
+    :codeauthor: :email:`Anthony Shaw <anthonyshaw@apache.org>`
+'''
+from __future__ import absolute_import
+from __future__ import print_function
+import yaml
```

Comment:
It might be nice to use the salt serializer classes here. We do some work to try to massage/optimize YAML. See `salt.serializers.yaml` for example.